### ISSUE TITLE
Separate fields instead of {name} and {adress}?

### ISSUE THREAD

**Issue by ilu33** [byro/byro#130]
Problem: We are an old club. We need to adress members either by first name or formally by last name, depending on the member. We can't just write "Hi!" to people who don't feel comfortable with that level of informality.
We can add the fields {firstname} and {lastname} with a plugin but byro will still require and use the {name} field which would result in redundant data. We also don't know how to best implement a mechanic that selects the member-specific greeting in emails (Hallo Peter or Sehr geehrter Herr Meier or Sehr geehrte Frau Meier).

Also our old database has separate fields for street, zip code, city, country and a field for addendums like "c/o" - we could join them into one field but then the c/o part would also show up in the donation receipt generated by the plugin byro-gemeinnuetzigkeit. Information gained from a c/o should not show up in documents for authorities, it's not our right to "out" anybody. Creating additional fields is easy but again leads to annoying redundancies.

Any advice?

**Comment by rixx** [comment]
I don't think we can split up these fields in byro itself – for one, the migration would be a headache for existing users, and it would also make a bunch of assumptions that would not hold true for all associations.

The correct approach would probably be to make the name field optional and allow plugins to provide the name, and same with the address. That might not even be a terribly big change within byro, and should be doable in a backwards compatible fashion.

**Comment by ilu33** [comment]
We thought about filling the name and address fields automatically with a plugin but had doubts about keeping this dependency consistent through all operations.

**Comment by rixx** [comment]
Yeah, I agree, that wouldn't work. But if we provided a set class interface or signal that plugins could use to provide address and name (read-only), that would work.
